drivers/storage/portworx/deployment_test.go

```diff
@@ -1904,12 +1966,238 @@ func TestPodSpecForKvdbAuthErrorReadingSecret(t *testing.T) {
+
+	// Storage devices and auto-compute max_storage_nodes_per_zone
+	cluster.Spec.CloudStorage = &corev1alpha1.CloudStorageSpec{
+		CapacitySpecs: []corev1alpha1.CloudStorageCapacitySpec{
+			{
+				MinIOPS:          uint32(100),
+				MinCapacityInGiB: uint64(100),
+				MaxCapacityInGiB: uint64(200),
+				Options:          map[string]string{"foo1": "bar1"},
+			},
+			{
+				MinIOPS:          uint32(200),
+				MinCapacityInGiB: uint64(200),
+				MaxCapacityInGiB: uint64(500),
+				Options:          map[string]string{"foo3": "bar3"},
+			},
+		},
+	}
+
+	expectedArgs = []string{
+		"-c", "px-cluster",
+		"-x", "kubernetes",
+		"-s", "type=foo,size=120,iops=110,foo1=bar1",
+		"-s", "type=bar,size=220,iops=210,foo3=bar3",
+		"-max_storage_nodes_per_zone", "2",
+	}
+
+	inputInstancesPerZone := 2
+	mockStorageManager.EXPECT().
+		GetStorageDistribution(&cloudops.StorageDistributionRequest{
+			ZoneCount:        len(zoneToInstancesMap),
+			InstancesPerZone: inputInstancesPerZone,
+			UserStorageSpec: []*cloudops.StorageSpec{
+				{
+					IOPS:        uint32(100),
+					MinCapacity: uint64(100),
+					MaxCapacity: uint64(200),
+				},
+				{
+					IOPS:        uint32(200),
+					MinCapacity: uint64(200),
+					MaxCapacity: uint64(500),
+				},
+			},
+		}).
+		Return(&cloudops.StorageDistributionResponse{
+			InstanceStorage: []*cloudops.StoragePoolSpec{
+				{
+					DriveCapacityGiB: uint64(120),
+					DriveType:        "foo",
+					DriveCount:       1,
+					InstancesPerZone: inputInstancesPerZone,
+					IOPS:             uint32(110),
+				},
+				{
+					DriveCapacityGiB: uint64(220),
+					DriveType:        "bar",
+					DriveCount:       1,
+					InstancesPerZone: inputInstancesPerZone,
+					IOPS:             uint32(210),
+				},
+			},
+		}, nil)
+
+	actual, err = driver.GetStoragePodSpec(cluster, nodeName)
+	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
+	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
+
+	list, err = driver.storageNodesList(cluster)
+	assert.NoError(t, err, "Unexpected error on GetStorageNodeList")
+	assert.Equal(t, len(list), 1, "expected storage nodes in list")
+
+	expectedCloudStorageSpec := []corev1alpha1.StorageNodeCloudDriveConfig{
+		{
+			Type:      "foo",
+			SizeInGiB: uint64(120),
+			IOPS:      uint32(110),
+			Options:   map[string]string{"foo1": "bar1"},
+		},
+		{
+			Type:      "bar",
+			SizeInGiB: uint64(220),
+			IOPS:      uint32(210),
+			Options: map[string]string{
+				"foo3": "bar3",
+			}},
+	}
+
+	expectedArgs = []string{
+		"-c", "px-cluster",
+		"-x", "kubernetes",
+		"-s", "type=foo,size=120,iops=110,foo1=bar1",
+		"-s", "type=bar,size=220,iops=210,foo3=bar3",
+		"-max_storage_nodes_per_zone", "2",
+	}
+
+	// despite adding a new node, node spec should be pulled from the existing storage node for nodeName
+	// Storage node will be created for "nodeName1"
+	actual, err = driver.GetStoragePodSpec(cluster, nodeName2)
```

Comment:
Here we should also increase the instancesPerZone map in the driver. Basically add more instances there. Just to make sure that it still returns the same output.

Reply:
As per my understanding, instancesPerZone should not be modified after a cluster has been initialized

Reply:
It keeps getting updated periodically. If more instances are added, it will have the latest `instancesPerZone counts`.